src/extensions/replay/sessionrecording.ts

```diff
@@ -459,28 +580,52 @@ export class SessionRecording {
         return url
     }
 
+    private clearBuffer(): SnapshotBuffer {
+        this.buffer = undefined
+
+        return {
+            size: 0,
+            data: [],
+            sessionId: this.sessionId,
+            windowId: this.windowId,
+        }
+    }
+
+    // the intention is a buffer that (currently) is used only after a decide response enables session recording
+    // it is called ever X seconds using the flushBufferTimer so that we don't have to wait for the buffer to fill up
+    // when it is called on a timer it assumes that it can definitely flush
+    // it is flushed when the session id changes or the size of the buffered data gets too great (1mb by default)
+    // first change: if the recording is in buffering mode,
+    //  flush buffer simply resets the timer and returns the existing flush buffer
     private _flushBuffer() {
         if (this.flushBufferTimer) {
             clearTimeout(this.flushBufferTimer)
             this.flushBufferTimer = undefined
         }
 
+        const minimumDuration = this.minimumDuration
+        const bufferedDuration = this.bufferedDuration
+        const isBelowMinimumDuration =
```

Comment:
This threw me as I thought "wait - this should be the sessionDuration!" and then I saw that actually it is the sessionDuration, it's just named as the buffer duration which feels wrong